tests/test_types.py

```diff
@@ -948,6 +948,24 @@ class Model(BaseModel):
         Model(v=b'foobar')
 
 
+def test_strict_str_subclass():
+    class MyStrictStr(StrictStr):
+        pass
+
+    class Model(BaseModel):
+        v: MyStrictStr
```

Comment:
In particular, could you add `class MyStrictInt(StrictInt): pass` (and similar for `StrictFloat` and `StrictBool`), add them as fields on `Model`, and pass appropriate values in the `m = Model(...` line?

No need to add any more `with pytest.raises(ValidationError)` checks, since those shouldn't be affected by this issue.